chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/RoomDeleted.kt

```diff
@@ -0,0 +1,26 @@
+package com.pusher.chatkit.rooms.state
+
+import com.pusher.chatkit.state.ChatkitState
+import org.reduxkotlin.reducerForActionType
+
+internal data class RoomDeleted(
+    val roomId: String
+)
+
+internal val roomDeletedReducer =
+    reducerForActionType<ChatkitState, RoomDeleted> { state, action ->
+
+            var joinedRoomsState = JoinedRoomsState(
+                    hashMapOf(),
+                    hashMapOf()
+            )
+
+            if (state.joinedRoomsState != null) {
+                joinedRoomsState = JoinedRoomsState(
+                        state.joinedRoomsState.rooms.filterNot { it.key == action.roomId },
```

Comment:
there's no need for iterating over the whole collection and doing the filtering to remove an entry, it's better (also more legible) to simply use the minus operator:
```suggestion
                        state.joinedRoomsState.rooms - action.roomId,
```

Reply:
👍